e2e/js_run_devserver/multirun_test.sh

```diff
@@ -14,107 +14,119 @@ _sedi () {
+    echo "$$: ERROR: Expected http://localhost:8081 to be available"
+    exit 1
+  fi
+  sleep 1 # wait before check again
+  ((n=n+1))
 done
 
-echo "Waiting 10 seconds for devservers to settle to mitigate flakiness..."
-sleep 10
-
-echo "Devservers ready"
+echo "$$: Devservers ready"
 
 if ! curl http://localhost:8080/index.html --fail 2>/dev/null | grep "My first website"; then
-  echo "ERROR: Expected http://localhost:8080/index.html to contain 'My first website'"
+  echo "$$: ERROR: Expected http://localhost:8080/index.html to contain 'My first website'"
   exit 1
 fi
 
 if ! curl http://localhost:8081/index.html --fail 2>/dev/null | grep "My first website"; then
-  echo "ERROR: Expected http://localhost:8080/index.html to contain 'My first website'"
+  echo "$$: ERROR: Expected http://localhost:8081/index.html to contain 'My first website'"
   exit 1
 fi
 
 if ! curl http://localhost:8080/other.html --fail 2>/dev/null | grep "My other website"; then
-  echo "ERROR: Expected http://localhost:8080/other.html to contain 'My other website'"
+  echo "$$: ERROR: Expected http://localhost:8080/other.html to contain 'My other website'"
   exit 1
 fi
 
 if ! curl http://localhost:8081/other.html --fail 2>/dev/null | grep "My other website"; then
-  echo "ERROR: Expected http://localhost:8080/other.html to contain 'My other website'"
+  echo "$$: ERROR: Expected http://localhost:8081/other.html to contain 'My other website'"
   exit 1
 fi
 
 if curl http://localhost:8080/new.html --fail 2>/dev/null; then
-  echo "ERROR: Expected http://localhost:8080/new.html to fail with 404"
+  echo "$$: ERROR: Expected http://localhost:8080/new.html to fail with 404"
   exit 1
 fi
 
 if curl http://localhost:8081/new.html --fail 2>/dev/null; then
-  echo "ERROR: Expected http://localhost:8080/new.html to fail with 404"
+  echo "$$: ERROR: Expected http://localhost:8081/new.html to fail with 404"
   exit 1
 fi
 
 if curl http://localhost:8080/index.html --fail 2>/dev/null | grep "A second line"; then
-  echo "ERROR: Expected http://localhost:8080/index.html to NOT contain 'A second line'"
+  echo "$$: ERROR: Expected http://localhost:8080/index.html to NOT contain 'A second line'"
   exit 1
 fi
 
 if curl http://localhost:8081/index.html --fail 2>/dev/null | grep "A second line"; then
-  echo "ERROR: Expected http://localhost:8080/index.html to NOT contain 'A second line'"
+  echo "$$: ERROR: Expected http://localhost:8081/index.html to NOT contain 'A second line'"
   exit 1
 fi
 
-echo "<div>A second line</div>" >> src/index.html
-
-echo "Waiting 10 seconds for ibazel rebuild after change to src/index.html..."
-sleep 10
+echo "$$: <div>A second line</div>" >> src/index.html
 
-if ! curl http://localhost:8080/index.html --fail 2>/dev/null | grep "A second line"; then
-  echo "ERROR: Expected http://localhost:8080/index.html to contain 'A second line'"
-  exit 1
-fi
+n=0
+while ! curl http://localhost:8080/index.html --fail 2>/dev/null | grep "A second line"; do
```

Comment:
The ugly loop avoids waiting 10s every single time. With the amount of times this script (and the other) are run it seemed worth copy+pasting the loop a few times 🤷